src/index.js

```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import { RouterProvider } from "react-router-dom";
+import { router } from "./router.jsx";
+import "./styles/index.css";
+
+ReactDOM.createRoot(document.getElementById("root")).render(
+  <React.StrictMode>
+    <RouterProvider router={router} />
```

Comment:
Such an approach prevents you from using the common app state for the different routes. It will render completely new component for every route including common parts as header/footer